pyfr/plugins/fluidforce.py

```diff
@@ -171,16 +211,26 @@ def __call__(self, intg):
                     vis = self.stress_tensor(ufpts, dufpts)
 
                 # Do the quadrature
-                f[ndims:] += np.einsum('i...,klij,jil', qwts, vis, norms)
+                fv += np.einsum('i...,klij,jil', qwts, vis, norms)
+                if self._mcomp:
+                    # Get the relative flux points position
+                    rfpts = self._rfpts[etype, fidx]
+
+                    # Viscous force at each flux point
+                    viscf = np.einsum('ijkl,lkj->lki', vis, norms)
+
+                    # Viscous force moments
+                    rcf = np.atleast_3d(np.cross(rfpts, viscf))
+                    mv += np.einsum('i,jik->k', qwts, rcf)
 
         # Reduce and output if we're the root rank
         if rank != root:
-            comm.Reduce(f, None, op=get_mpi('sum'), root=root)
+            comm.Reduce(fm, None, op=get_mpi('sum'), root=root)
         else:
-            comm.Reduce(get_mpi('in_place'), f, op=get_mpi('sum'), root=root)
+            comm.Reduce(get_mpi('in_place'), fm, op=get_mpi('sum'), root=root)
 
             # Write
-            print(intg.tcurr, *f, sep=',', file=self.outf)
+            print(intg.tcurr, *fp, *fv, *mp, *mv, sep=',', file=self.outf)
```

Comment:
`f.ravel` might be able to help us out here (may need to refine the shape of f to be its transpose, though).